spec/controllers/admin/products_controller_spec.rb

```diff
@@ -0,0 +1,119 @@
+require "rails_helper"
+
+RSpec.describe Admin::ProductsController, type: :controller do
+  let!(:brand) { FactoryGirl.create(:brand) }
+  let!(:category) { FactoryGirl.create(:category) }
+
+  let(:user) { FactoryGirl.create(:admin) }
+
+  before { sign_in_as(user) }
+
+  let(:valid_attributes) {
+    { name: "Foo", brand_id: brand.id, category_id: category.id, price: 10.99 }
+  }
+
+  let(:invalid_attributes) {
+    { name: "", price: -50 }
+  }
+
+  let(:valid_session) { {} }
+
+  describe "GET #index" do
+    it "assigns all products as @products" do
+      product = Product.create! valid_attributes
+      get :index, {}, valid_session
+      expect(assigns(:products)).to eq([product])
+    end
+  end
+
+  describe "GET #new" do
+    it "assigns a new product as @product" do
+      get :new, {}, valid_session
+      expect(assigns(:product)).to be_a_new(Product)
+    end
+  end
+
+  describe "GET #edit" do
+    it "assigns the requested product as @product" do
+      product = Product.create! valid_attributes
+      get :edit, {:id => product.to_param}, valid_session
+      expect(assigns(:product)).to eq(product)
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new Product" do
+        expect {
+          post :create, {:product => valid_attributes}, valid_session
+        }.to change(Product, :count).by(1)
+      end
+
+      it "assigns a newly created product as @product" do
+        post :create, {:product => valid_attributes}, valid_session
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.